engine/src/main/java/org/terasology/engine/modes/loadProcesses/PostBeginSystems.java

```diff
@@ -46,20 +44,18 @@ public String getMessage() {
     @Override
     public boolean step() {
         if (componentSystems.hasNext()) {
-            try {
-                currentSystem = componentSystems.next();
-                currentSystem.postBegin();
-            } catch (NoSuchElementException e) {
-                logger.error("Failed to load system : '" + currentSystem.toString() + "'");
-            }
+            componentSystems.next().postBegin();
+            stepDone();
         }
         return !componentSystems.hasNext();
     }
 
     @Override
     public void begin() {
-        ComponentSystemManager csm = context.get(ComponentSystemManager.class);
-        componentSystems = csm.iterateAll().iterator();
+        ComponentSystemManager manager = context.get(ComponentSystemManager.class);
+        final List<ComponentSystem> componentSystemList = manager.iterateAll();
+        componentSystems = componentSystemList.iterator();
+        setTotalSteps(componentSystemList.size());
```

Comment:
Ah, this is why you changed from `Iterable` to `List`? To have the `size` information?